File: C#/LAB2/StringProcessor.cs
using System;
using System.Linq;

namespace ConsoleApp4;

public class StringProcessor
{
    private string value;

    public StringProcessor()              
    {
        value = "";
    }

    public StringProcessor(string str)    
    {
        value = str;
    }

    ~StringProcessor()                    
    {
        // Деструктор
    }

    public int GetLength()
    {
        return value.Length;
    }

    public string GetString()
    {
        return value;
    }

    public void SortDescending()          
    {
        value = new string(value.OrderByDescending(c => c).ToArray());
    }
}

File: C++/LAB2/StringProcessor.cpp
#include "StringProcessor.h"
#include <algorithm>
#include <functional>

StringProcessor::StringProcessor()   
{
    value = "";
}

StringProcessor::StringProcessor(std::string str)
{
    value = str;
}

StringProcessor::~StringProcessor()  
{
    // Деструктор
}

int StringProcessor::GetLength() const
{
    return value.length();
}

std::string StringProcessor::GetString() const
{
    return value;
}

void StringProcessor::SortDescending()   
{
    std::sort(value.begin(), value.end(), std::greater<char>());
}